docs/source/examples/advanced_error_mitigation_pipeline.md

```diff
@@ -0,0 +1,704 @@
+                           p_readout_param=base_p_readout,
+                           depol_prob_param=base_depol_prob)
+    
+    if verbose:
+        print(f"\n[1/3] Applying REM with parameters: p0={base_p_readout}, p1={base_p_readout}")
+    
+    # 1. Apply REM to the base executor
+    rem_exec = rem.mitigate_executor(
+        base_noisy_exec, 
+        inverse_confusion_matrix=inverse_confusion_matrix
+    )
+    
+    if verbose:
+        print(f"\n[2/3] Applying DDD with rule: {ddd_rule.__name__}")
+    
+    # 2. Apply DDD to the circuit
+    ddd_circuit = ddd.insert_ddd_sequences(circuit_to_run, ddd_rule)
+    
+    if verbose:
+        print(f"Circuit depth after DDD: {len(ddd_circuit)}")
+    
+    if verbose:
+        print(f"\n[3/3] Applying PT with {num_twirled_variants} twirling variants")
+    
+    # 3. Apply PT to the DDD-modified circuit
+    pt_variants = generate_pauli_twirl_variants(
+        ddd_circuit, 
+        num_circuits=num_twirled_variants,
+        random_state=0  # Fixed seed for reproducibility
+    )
+    
+    # Execute all PT+DDD variants with REM and average the results
+    expectations = []
+    
+    if verbose:
+        print(f"\nExecuting {num_twirled_variants} twirled circuits with REM-wrapped executor:")
+        
+    for i, c in enumerate(pt_variants):
+        if verbose:
+            print(f"  Executing PT variant {i+1}/{num_twirled_variants}...", end="")
+        
+        exp_val = obs.expectation(c, rem_exec).real
+        expectations.append(exp_val)
+        
+        if verbose:
+            print(f" exp_val = {exp_val:.6f}")
+    
+    result = np.mean(expectations)
+    
+    if verbose:
+        print(f"\nFinal expectation value (averaged over {num_twirled_variants} PT variants): {result:.6f}")
+        print(f"{'='*60}")
+    
+    return result
+
+# Set up ZNE with circuit folding
+pipeline_scale_factors = [1.0, 1.5, 2.0]
+pipeline_zne_factory = LinearFactory(scale_factors=pipeline_scale_factors)
+
+# Create the ZNE mitigated executor
+full_pipeline_zne_executor_approach1 = zne.mitigate_executor(
+    executor=full_mitigation_executor,  # This executor already applies PT+DDD+REM
+    observable=None,  # We don't need to pass an observable as full_mitigation_executor returns a float
+    scale_noise=zne.scaling.folding.fold_global,  # Scale noise by folding the circuit
+    factory=pipeline_zne_factory
+)
+
+# Execute the pipeline
+full_pipeline_result_val_approach1 = full_pipeline_zne_executor_approach1(circuit)
+if hasattr(full_pipeline_result_val_approach1, 'real'):
+    full_pipeline_result_val_approach1 = full_pipeline_result_val_approach1.real
+
+print(f"Approach 1 Full pipeline (PT→DDD→REM→ZNE via circuit folding) result: {full_pipeline_result_val_approach1:.6f}")
+print(f"Approach 1 Full pipeline absolute error: {abs(ideal_result_val - full_pipeline_result_val_approach1):.6f}")
+```
+
+### Approach 2: Structured Two-Stage Application (Conceptual for ZNE)
```

Comment:
Applying the full pipeline should only happen once. No need to go through the whole process twice.